schema/spacewalk/common/views/rhnServerOverview.sql

```diff
@@ -109,7 +110,8 @@ select
     os,
     release,
     ( select name from rhnServerArch where id = s.server_arch_id),
-    coalesce((select 1 from rhnServerLock SL WHERE SL.server_id = S.id), 0)
+    coalesce((select 1 from rhnServerLock SL WHERE SL.server_id = S.id), 0),
+    ( select pxy.server_Id from rhnProxyInfo pxy where pxy.server_id = S.id)
```

Comment:
I think you need to use coalesce here as well. In case it is not a proxy you would get "NULL".
As far as I can see, the jsp did not work wil NULL value here. Or did you test it?
Do you really need the ID, or just a "yes/no". As you use this value with `PROXY_ID AS IS_RHN_PROXY`
it sounds like 1/0 would be enough.

Reply:
Yes, I tested, the screenshot was taken from a properly working setup with this patch. The reason why I fetch the info in this way is because it was already designed to be like that, you can see it [here](https://github.com/uyuni-project/uyuni/blob/master/java/code/src/com/redhat/rhn/frontend/dto/SystemOverview.java#L165) : this is the DTO used to be populated by the `rhnServerOverview` view called from [here](https://github.com/uyuni-project/uyuni/blob/master/java/code/src/com/redhat/rhn/frontend/action/systems/SystemListSetupAction.java#L34)

Reply:
ok